onnx/defs/tensor/defs.cc

```diff
@@ -156,7 +250,38 @@ ONNX_OPERATOR_SCHEMA(Split)
     .SetDoc(R"DOC(Split a tensor into a list of tensors, along the specified
 'axis'. Lengths of the parts can be specified using argument 'split'.
 Otherwise, the tensor is split to equal sized parts.
-)DOC");
+)DOC")
+    .ShapeInferenceFunction([](InferenceContext& ctx) {
+        propagateElemTypeFromInputToOutput(ctx, 0, 0);
+
+        if (ctx.getNumOutputs() == 0) {
+          return;
+        }
+
+        auto axisAttr = ctx.getAttribute("axis");
+        int axis = axisAttr ? axisAttr->i() : 0;
+        std::vector<int64_t> split;
+        if (!getRepeatedAttribute(ctx, "split", split)) {
+          if (!ctx.getInputType(0)->has_shape()) {
+            return;
+          }
+          const auto& splitDim = ctx.getInputType(0)->shape().dim(axis);
+          if (!splitDim.has_dim_value()) {
+            return;
+          }
+          int splitDimValue = splitDim.dim_value();
+          int chunkSize = splitDimValue / ctx.getNumOutputs();
+          int leftOver = splitDimValue - (chunkSize * ctx.getNumOutputs());
+          for (int i = 0; i < ctx.getNumOutputs(); i++) {
+            split.push_back(i < leftOver ? chunkSize + 1 : chunkSize);
```

Comment:
The spec above doesn't actually say what the behavior is when leftover is non-zero. May be a good idea to extend the spec to do that (though that should probably be a separate PR since it affects the spec).

Reply:
++ will make a PR for the spec